state/core_access.go

```diff
@@ -214,6 +238,13 @@ func (ca *CoreAccessor) SubmitPayForBlob(
 
 	minGasPrice := ca.getMinGasPrice()
 
+	var feeGrant apptypes.TxBuilderOption
+
+	// set granter and update gasLimit in case node run in a grantee mode
```

Comment:
Why do we need the gas multiplier here?

Reply:
gasLimit for tx with options is higher the without options, but app does not include in the calculations the options cost. So me and @cmwaters agreed to use a multiplier as a quick solution.